### DISCUSSION TITLE
Multiple Buttons Controlling Multiple LED's using 2.Button.ino

### DISCUSSION THREAD

**Discussion by hali1850** [tttapa/Control-Surface#729] [Q&A] [answered]
Hi,

I am very new to coding and to programming my Teensy LC. 
I am trying to add multiple buttons (2 for now) each with corresponding LED's, which turn on when I press the corresponding button, and off when I press again- exactly as is the function of the '2.Button.ino' example, except I would like to use multiple buttons with associated LED's (5, ultimately), instead of just the 1 as in the example. 

I have 2 buttons hooked up to pins 2 and 4 respectively, and 2 LED's hooked up to pins 20 and 19 respectively. My breadboard is a little cluttered so this the setup for now. 

(Also important to mention: The primary function of each button is to be a CCButton to send CC information into a DAW, rather than to just simply toggle an LED. Is it possible for the code to accommodate both of these features? And if so, how?)

Here is the code which won't compile of course. 

`// Include the library
#include <Arduino_Helpers.h>
 
#include <AH/Hardware/Button.hpp>
 
// Create a Button object that reads a push button connected to pin 2:
Button pushbutton {2, 4};
 
// The pin with the LED connected:
const pin_t ledPin = {20, 19};

void setup() {
  pinMode(ledPin, OUTPUT);
  pushbutton.begin();
  // You can invert the input, for use with normally closed (NC) switches:
  // pushbutton.invert();
}
 
void loop() {
  static bool ledState = LOW;
  // Read the digital input, debounce the signal, and check the state of
  // the button:
  if (pushbutton.update() == Button::Falling) {
    ledState = !ledState; // Invert the state of the LED
    // Update the LED with the new state
    digitalWrite(ledPin, ledState ? HIGH : LOW);
  }
}` 

So far, as I'm sure you know, only the first button and LED pair work properly, while the second does not function. 
How can I get this to work, and how should I alter the code to enable this? Any help at all is really appreciated. 

Sidenote: While I am just trying to get this simple setup up and running for prototyping purposes, my ultimate goal is actually to have these LED's 'ambiently' backlight each button by default when the device is turned on. But when a button is pressed, the associated LED glows brighter to indicate that the buttons function is in effect. And conversely, when the button is again pressed, the LED then reverts back to its ambient/lower brightness. Could someone here kindly let me know a) how this would be achieved, b) if it is achievable using this Control Surface repository and if so, how? 

Regards!

**Comment by tttapa** [comment, marked as answer]
For the CCButton + LED, the code I posted here should get you started: https://github.com/tttapa/Control-Surface/issues/338#issuecomment-733151457
You can make it toggle on/off by keeping a boolean flag like the `ledState` variable like you have in your code above.

> my ultimate goal is actually to have these LED's 'ambiently' backlight each button by default when the device is turned on. But when a button is pressed, the associated LED glows brighter to indicate that the buttons function is in effect. And conversely, when the button is again pressed, the LED then reverts back to its ambient/lower brightness. Could someone here kindly let me know a) how this would be achieved, b) if it is achievable using this Control Surface repository and if so, how?

Sure, you can easily do this in hardware, by having a high-valued resistor that always turns on the LED a little bit, and then use `digitalWrite()` in software to turn it fully on. Alternatively, you could use PWM to do it all in software. Both methods can be used in combination with the Control Surface library.
